### PR TITLE
feature (api): create simple fastapi server

### PR DESCRIPTION
**Pull request by bukkalexander**
Created a FastAPI server and connected the frontend to it locally.

### CODE REVIEW

**Review by mghnm — COMMENTED** [review]
The use of fast_api to deploy a webserver might not be appropriate when using api_gateway and lambda. This is for the reason that api_gateway in this case acts as a proxy for lambda. Every time it receives a request it will invoke lambda.

**Comment by bukkalexander** [comment]
You are right, but there is a workaround using a library called 'mangum' that will make it work inside a lambda automatically, according to chatgpt at least 😅 But we can use plain functions also, whatever feels best

**Comment by bukkalexander** [comment]
The only reason I'd like to use fastapi is that I wanted to learn how to use it, and it will be simple (I think) to use it both locally when developing and remotely when deployed as lambda, since the routing can be tested locally, and when deployed, we can configure it to use mangum to handle the event and convert it to the fastapi interface. Also, if we decide to switch to ec2, we already get an easy api server setup for free.

Example

```

from fastapi import FastAPI
from mangum import Mangum

app = FastAPI()

@app.get("/")
def read_root():
    return {"message": "Hello from Lambda!"}

# Add Mangum handler to work with AWS Lambda
handler = Mangum(app)

````

**Comment by mghnm** [comment]
Application main scenario is in working state. FE is communicating with BE and fetching the test questions from the APP. Next step would be throwing those questions into the DynamoDB table for complete integration.

File: web/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "@emotion/react": "^11.13.3",
     "@emotion/styled": "^11.13.0",
     "@mui/material": "^6.1.4",
+    "dotenv": "^16.4.5",
```
Comment (ifayazagreat) [inline web/package.json]: remove

**Review by ifayazagreat — COMMENTED** [review]
see review comment

**Review by ifayazagreat — APPROVED** [review]

**Pull request merged**